scipy/special/_logsumexp.py

```diff
@@ -202,7 +202,10 @@ def _logsumexp(a, b, axis, return_sign, xp):
     a_max, i_max = _elements_and_indices_with_max_real(a, axis=axis, xp=xp)
 
     # for precision, these terms are separated out of the main sum.
-    a[i_max] = -xp.inf
+    # TODO: we shouldn't be mutating in-place here unless we make a copy
+    # dask arrays do not copy before this somehow
+    #a[i_max] = -xp.inf
+    a = xp.where(i_max, -xp.asarray(xp.inf, dtype=a.dtype), a)
```

Comment:
just flagging this as a review comment so it doesn't get lost

Reply:
A copy is always made, due to this line near the top of  the `logsumexp` implementation:
```python
a, b = xp_broadcast_promote(a, b, ensure_writeable=True, ...
```

So this will yield correct semantics. It may be slower (didn't check) since a copy is always made now, while the previous implementation (e.g., in v1.14.1) only made a copy if `np.any(b == 0)` is True.

Potentially, not copying and avoiding any in-place mutation but using `xp.where` is therefore beneficial also for performance with `numpy` in the common case where `b` isn't used or doesn't have elements that are `0`.